diff --git a/arch/arm64/boot/dts/marvell/accton-as5114.dts b/arch/arm64/boot/dts/marvell/accton-as5114.dts
new file mode 100644
index 000000000..67326ff44
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/accton-as5114.dts
@@ -0,0 +1,1864 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include "armada-7040.dtsi"
+
+/ {
+	model = "Accton Marvell Armada 7040 board setup";
+	compatible = "marvell,armada-7040-as4224",
+		     "marvell,armada7040", "marvell,armada-ap806-quad",
+		     "marvell,armada-ap806";
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
+		i2c0 = &i2c0;
+		i2c1 = &cp0_i2c0;
+		i2c2 = &cp0_i2c1;
+	};
+
+	switch-cpu {
+		compatible = "marvell,prestera-switch-rxtx-sdma";
+		status = "okay";
+	};
+
+	onie_eeprom: onie-eeprom {
+		compatible = "onie-nvmem-cells";
+		status = "okay";
+		nvmem = <&eeprom_at24>;
+	};
+
+	prestera {
+		compatible = "marvell,prestera";
+		status = "okay";
+		base-mac-provider = <&onie_eeprom>;
+		ports {
+			port1 {
+				prestera,port-num = <1>;
+				sfp = <&sfp1>;
+			};
+			port2 {
+				prestera,port-num = <2>;
+				sfp = <&sfp2>;
+			};
+			port3 {
+				prestera,port-num = <3>;
+				sfp = <&sfp3>;
+			};
+			port4 {
+				prestera,port-num = <4>;
+				sfp = <&sfp4>;
+			};
+			port5 {
+				prestera,port-num = <5>;
+				sfp = <&sfp5>;
+			};
+			port6 {
+				prestera,port-num = <6>;
+				sfp = <&sfp6>;
+			};
+			port7 {
+				prestera,port-num = <7>;
+				sfp = <&sfp7>;
+			};
+			port8 {
+				prestera,port-num = <8>;
+				sfp = <&sfp8>;
+			};
+			port9 {
+				prestera,port-num = <9>;
+				sfp = <&sfp9>;
+			};
+			port10 {
+				prestera,port-num = <10>;
+				sfp = <&sfp10>;
+			};
+			port11 {
+				prestera,port-num = <11>;
+				sfp = <&sfp11>;
+			};
+			port12 {
+				prestera,port-num = <12>;
+				sfp = <&sfp12>;
+			};
+			port13 {
+				prestera,port-num = <13>;
+				sfp = <&sfp13>;
+			};
+			port14 {
+				prestera,port-num = <14>;
+				sfp = <&sfp14>;
+			};
+			port15 {
+				prestera,port-num = <15>;
+				sfp = <&sfp15>;
+			};
+			port16 {
+				prestera,port-num = <16>;
+				sfp = <&sfp16>;
+			};
+			port17 {
+				prestera,port-num = <17>;
+				sfp = <&sfp17>;
+			};
+			port18 {
+				prestera,port-num = <18>;
+				sfp = <&sfp18>;
+			};
+			port19 {
+				prestera,port-num = <19>;
+				sfp = <&sfp19>;
+			};
+			port20 {
+				prestera,port-num = <20>;
+				sfp = <&sfp20>;
+			};
+			port21 {
+				prestera,port-num = <21>;
+				sfp = <&sfp21>;
+			};
+			port22 {
+				prestera,port-num = <22>;
+				sfp = <&sfp22>;
+			};
+			port23 {
+				prestera,port-num = <23>;
+				sfp = <&sfp23>;
+			};
+			port24 {
+				prestera,port-num = <24>;
+				sfp = <&sfp24>;
+			};
+			port25 {
+				prestera,port-num = <25>;
+				sfp = <&sfp25>;
+			};
+			port26 {
+				prestera,port-num = <26>;
+				sfp = <&sfp26>;
+			};
+			port27 {
+				prestera,port-num = <27>;
+				sfp = <&sfp27>;
+			};
+			port28 {
+				prestera,port-num = <28>;
+				sfp = <&sfp28>;
+			};
+			port29 {
+				prestera,port-num = <29>;
+				sfp = <&sfp29>;
+			};
+			port30 {
+				prestera,port-num = <30>;
+				sfp = <&sfp30>;
+			};
+			port31 {
+				prestera,port-num = <31>;
+				sfp = <&sfp31>;
+			};
+			port32 {
+				prestera,port-num = <32>;
+				sfp = <&sfp32>;
+			};
+			port33 {
+				prestera,port-num = <33>;
+				sfp = <&sfp33>;
+			};
+			port34 {
+				prestera,port-num = <34>;
+				sfp = <&sfp34>;
+			};
+			port35 {
+				prestera,port-num = <35>;
+				sfp = <&sfp35>;
+			};
+			port36 {
+				prestera,port-num = <36>;
+				sfp = <&sfp36>;
+			};
+			port37 {
+				prestera,port-num = <37>;
+				sfp = <&sfp37>;
+			};
+			port38 {
+				prestera,port-num = <38>;
+				sfp = <&sfp38>;
+			};
+			port39 {
+				prestera,port-num = <39>;
+				sfp = <&sfp39>;
+			};
+			port40 {
+				prestera,port-num = <40>;
+				sfp = <&sfp40>;
+			};
+			port41 {
+				prestera,port-num = <41>;
+				sfp = <&sfp41>;
+			};
+			port42 {
+				prestera,port-num = <42>;
+				sfp = <&sfp42>;
+			};
+			port43 {
+				prestera,port-num = <43>;
+				sfp = <&sfp43>;
+			};
+			port44 {
+				prestera,port-num = <44>;
+				sfp = <&sfp44>;
+			};
+			port45 {
+				prestera,port-num = <45>;
+				sfp = <&sfp45>;
+			};
+			port46 {
+				prestera,port-num = <46>;
+				sfp = <&sfp46>;
+			};
+			port47 {
+				prestera,port-num = <47>;
+				sfp = <&sfp47>;
+			};
+			port48 {
+				prestera,port-num = <48>;
+				sfp = <&sfp48>;
+			};
+		};
+	};
+
+	sfp1: sfp-1 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp0>;
+
+		/*
+			The below three lines is intended to support the SFP cage interfaces:
+				los-gpio - RX_LOSS
+				mod-def0-gpio - PRESENT_BIT
+				tx-disable-gpio - TX_ENABLE (or DISABLE)
+			To support these three signal the driver 'gpio_i2c' (Or compatible platform driver) must be enabled and supported.
+			Please refer to the 'gpio_i2c' section on this DTS file below.
+			It is currently not required for Amazon 'ethtool -m' support but it is intended for future use.
+			Can be skipped in this stage.
+		*/
+		los-gpio = <&gpio_i2c 0 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 1 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 2 GPIO_ACTIVE_HIGH>;
+	};
+	sfp2: sfp-2 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp1>;
+
+		los-gpio = <&gpio_i2c 3 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 4 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 5 GPIO_ACTIVE_HIGH>;
+	};
+	sfp3: sfp-3 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp2>;
+
+		los-gpio = <&gpio_i2c 6 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 7 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 8 GPIO_ACTIVE_HIGH>;
+	};
+	sfp4: sfp-4 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp3>;
+
+		los-gpio = <&gpio_i2c 9 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 10 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 11 GPIO_ACTIVE_HIGH>;
+	};
+	sfp5: sfp-5 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp4>;
+
+		los-gpio = <&gpio_i2c 12 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 13 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 14 GPIO_ACTIVE_HIGH>;
+	};
+	sfp6: sfp-6 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp5>;
+
+		los-gpio = <&gpio_i2c 15 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 16 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 17 GPIO_ACTIVE_HIGH>;
+	};
+	sfp7: sfp-7 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp6>;
+
+		los-gpio = <&gpio_i2c 18 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 19 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 20 GPIO_ACTIVE_HIGH>;
+	};
+	sfp8: sfp-8 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp7>;
+
+		los-gpio = <&gpio_i2c 21 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 22 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 23 GPIO_ACTIVE_HIGH>;
+	};
+	sfp9: sfp-9 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp8>;
+
+		los-gpio = <&gpio_i2c 24 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 25 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 26 GPIO_ACTIVE_HIGH>;
+	};
+	sfp10: sfp-10 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp9>;
+
+		los-gpio = <&gpio_i2c 27 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 28 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 29 GPIO_ACTIVE_HIGH>;
+	};
+	sfp11: sfp-11 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp10>;
+
+		los-gpio = <&gpio_i2c 30 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 31 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 32 GPIO_ACTIVE_HIGH>;
+	};
+	sfp12: sfp-12 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp11>;
+
+		los-gpio = <&gpio_i2c 33 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 34 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 35 GPIO_ACTIVE_HIGH>;
+	};
+	sfp13: sfp-13 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp12>;
+
+		los-gpio = <&gpio_i2c 36 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 37 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 38 GPIO_ACTIVE_HIGH>;
+	};
+	sfp14: sfp-14 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp13>;
+
+		los-gpio = <&gpio_i2c 39 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 40 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 41 GPIO_ACTIVE_HIGH>;
+	};
+	sfp15: sfp-15 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp14>;
+
+		los-gpio = <&gpio_i2c 42 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 43 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 44 GPIO_ACTIVE_HIGH>;
+	};
+	sfp16: sfp-16 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp15>;
+
+		los-gpio = <&gpio_i2c 45 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 46 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 47 GPIO_ACTIVE_HIGH>;
+	};
+	sfp17: sfp-17 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp16>;
+
+		los-gpio = <&gpio_i2c 48 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 49 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 50 GPIO_ACTIVE_HIGH>;
+	};
+	sfp18: sfp-18 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp17>;
+
+		los-gpio = <&gpio_i2c 51 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 52 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 53 GPIO_ACTIVE_HIGH>;
+	};
+	sfp19: sfp-19 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp18>;
+
+		los-gpio = <&gpio_i2c 54 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 55 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 56 GPIO_ACTIVE_HIGH>;
+	};
+	sfp20: sfp-20 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp19>;
+
+		los-gpio = <&gpio_i2c 57 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 58 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 59 GPIO_ACTIVE_HIGH>;
+	};
+	sfp21: sfp-21 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp20>;
+
+		los-gpio = <&gpio_i2c 60 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 61 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 62 GPIO_ACTIVE_HIGH>;
+	};
+	sfp22: sfp-22 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp21>;
+
+		los-gpio = <&gpio_i2c 63 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 64 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 65 GPIO_ACTIVE_HIGH>;
+	};
+	sfp23: sfp-23 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp22>;
+
+		los-gpio = <&gpio_i2c 66 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 67 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 68 GPIO_ACTIVE_HIGH>;
+	};
+	sfp24: sfp-24 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp23>;
+
+		los-gpio = <&gpio_i2c 69 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 70 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 71 GPIO_ACTIVE_HIGH>;
+	};
+	sfp25: sfp-25 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp24>;
+
+		los-gpio = <&gpio_i2c 72 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 73 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 74 GPIO_ACTIVE_HIGH>;
+	};
+	sfp26: sfp-26 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp25>;
+
+		los-gpio = <&gpio_i2c 75 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 76 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 77 GPIO_ACTIVE_HIGH>;
+	};
+	sfp27: sfp-27 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp26>;
+
+		los-gpio = <&gpio_i2c 78 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 79 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 80 GPIO_ACTIVE_HIGH>;
+	};
+	sfp28: sfp-28 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp27>;
+
+		los-gpio = <&gpio_i2c 81 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 82 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 83 GPIO_ACTIVE_HIGH>;
+	};
+	sfp29: sfp-29 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp28>;
+
+		los-gpio = <&gpio_i2c 84 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 85 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 86 GPIO_ACTIVE_HIGH>;
+	};
+	sfp30: sfp-30 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp29>;
+
+		los-gpio = <&gpio_i2c 87 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 88 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 89 GPIO_ACTIVE_HIGH>;
+	};
+	sfp31: sfp-31 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp30>;
+
+		los-gpio = <&gpio_i2c 90 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 91 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 92 GPIO_ACTIVE_HIGH>;
+	};
+	sfp32: sfp-32 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp31>;
+
+		los-gpio = <&gpio_i2c 93 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 94 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 95 GPIO_ACTIVE_HIGH>;
+	};
+	sfp33: sfp-33 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp32>;
+
+		los-gpio = <&gpio_i2c 96 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 97 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 98 GPIO_ACTIVE_HIGH>;
+	};
+	sfp34: sfp-34 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp33>;
+
+		los-gpio = <&gpio_i2c 99 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 100 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 101 GPIO_ACTIVE_HIGH>;
+	};
+	sfp35: sfp-35 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp34>;
+
+		los-gpio = <&gpio_i2c 102 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 103 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 104 GPIO_ACTIVE_HIGH>;
+	};
+	sfp36: sfp-36 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp35>;
+
+		los-gpio = <&gpio_i2c 105 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 106 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 107 GPIO_ACTIVE_HIGH>;
+	};
+	sfp37: sfp-37 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp36>;
+
+		los-gpio = <&gpio_i2c 108 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 109 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 110 GPIO_ACTIVE_HIGH>;
+	};
+	sfp38: sfp-38 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp37>;
+
+		los-gpio = <&gpio_i2c 111 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 112 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 113 GPIO_ACTIVE_HIGH>;
+	};
+	sfp39: sfp-39 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp38>;
+
+		los-gpio = <&gpio_i2c 114 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 115 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 116 GPIO_ACTIVE_HIGH>;
+	};
+	sfp40: sfp-40 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp39>;
+
+		los-gpio = <&gpio_i2c 117 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 118 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 119 GPIO_ACTIVE_HIGH>;
+	};
+	sfp41: sfp-41 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp40>;
+
+		los-gpio = <&gpio_i2c 120 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 121 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 122 GPIO_ACTIVE_HIGH>;
+	};
+	sfp42: sfp-42 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp41>;
+
+		los-gpio = <&gpio_i2c 123 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 124 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 125 GPIO_ACTIVE_HIGH>;
+	};
+	sfp43: sfp-43 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp42>;
+
+		los-gpio = <&gpio_i2c 126 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 127 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 128 GPIO_ACTIVE_HIGH>;
+	};
+	sfp44: sfp-44 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp43>;
+
+		los-gpio = <&gpio_i2c 129 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 130 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 131 GPIO_ACTIVE_HIGH>;
+	};
+	sfp45: sfp-45 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp44>;
+
+		los-gpio = <&gpio_i2c 132 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 133 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 134 GPIO_ACTIVE_HIGH>;
+	};
+	sfp46: sfp-46 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp45>;
+
+		los-gpio = <&gpio_i2c 135 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 136 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 137 GPIO_ACTIVE_HIGH>;
+	};
+	sfp47: sfp-47 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp46>;
+
+		los-gpio = <&gpio_i2c 138 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 139 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 140 GPIO_ACTIVE_HIGH>;
+	};
+	sfp48: sfp-48 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp47>;
+
+		los-gpio = <&gpio_i2c 141 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 142 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 143 GPIO_ACTIVE_HIGH>;
+	};
+
+	/*
+		This driver is intended to emulate GPIO interfaces on CPLD logic accessed via I2C interface.
+		It is required only if the SFP Cage interfaces will be required:
+				los-gpio - RX_LOSS
+				mod-def0-gpio - PRESENT_BIT
+				tx-disable-gpio - TX_ENABLE (or DISABLE)
+		This driver is added as reference code and can be replaced with another Platform Drivers that enables similar mapping.
+		It is currently not required for Amazon 'ethtool -m' support but it is intended for future use.
+		Can be skipped in this stage.
+	*/
+	gpio_i2c: gpio-i2c/* @40 */ {
+		compatible = "gpio-i2c";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		/* reg = <0x40>; */  /* CPLD/MUX I2C address */
+
+		gpio-map {
+			/* sfp0 */
+			sfp00_gpio00_loss {
+				reg-map = <0xA6 (1 << 0)>; /* 0xA6=register in the CPLD, (1 << 0)=Selected bit */
+				gpio-num = <0>;  /* Logical number used by: los-gpio, mod-def0-gpio and tx-disable-gpio */
+			};
+			sfp00_gpio01_pres {
+				reg-map = <0xC0 (1 << 0)>; /* reg mask */
+				gpio-num = <1>;
+			};
+			sfp00_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 0)>; /* reg mask */
+				gpio-num = <2>;
+			};
+
+			/* sfp1 */
+			sfp01_gpio00_sfp_loss {
+				reg-map = <0xA6 (1 << 1)>; /* reg mask */
+				gpio-num = <3>;
+			};
+			sfp01_gpio01_sfp_pres {
+				reg-map = <0xC0 (1 << 1)>; /* reg mask */
+				gpio-num = <4>;
+			};
+			sfp01_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 1)>; /* reg mask */
+				gpio-num = <5>;
+			};
+
+			/* sfp2 */
+			sfp02_gpio00_loss {
+				reg-map = <0xA6 (1 << 2)>; /* reg mask */
+				gpio-num = <6>;
+			};
+			sfp02_gpio01_pres {
+				reg-map = <0xC0 (1 << 2)>; /* reg mask */
+				gpio-num = <7>;
+			};
+			sfp02_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 2)>; /* reg mask */
+				gpio-num = <8>;
+			};
+
+			/* sfp3 */
+			sfp03_gpio01_loss {
+				reg-map = <0xA6 (1 << 3)>; /* reg mask */
+				gpio-num = <9>;
+			};
+			sfp03_gpio02_sfp_pres {
+				reg-map = <0xC0 (1 << 3)>; /* reg mask */
+				gpio-num = <10>;
+			};
+			sfp03_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 3)>; /* reg mask */
+				gpio-num = <11>;
+			};
+
+			/* sfp4 */
+			sfp04_gpio01_loss {
+				reg-map = <0xA6 (1 << 4)>; /* reg mask */
+				gpio-num = <12>;
+			};
+			sfp04_gpio02_sfp_pres {
+				reg-map = <0xC0 (1 << 4)>; /* reg mask */
+				gpio-num = <13>;
+			};
+			sfp04_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 4)>; /* reg mask */
+				gpio-num = <14>;
+			};
+
+			/* sfp5 */
+			sfp05_gpio01_loss {
+				reg-map = <0xA6 (1 << 5)>; /* reg mask */
+				gpio-num = <15>;
+			};
+			sfp05_gpio02_sfp_pres {
+				reg-map = <0xC0 (1 << 5)>; /* reg mask */
+				gpio-num = <16>;
+			};
+			sfp05_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 5)>; /* reg mask */
+				gpio-num = <17>;
+			};
+
+			/* sfp6 */
+			sfp06_gpio01_loss {
+				reg-map = <0xA6 (1 << 6)>; /* reg mask */
+				gpio-num = <18>;
+			};
+			sfp06_gpio02_sfp_pres {
+				reg-map = <0xC0 (1 << 6)>; /* reg mask */
+				gpio-num = <19>;
+			};
+			sfp06_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 6)>; /* reg mask */
+				gpio-num = <20>;
+			};
+
+			/* sfp7 */
+			sfp07_gpio01_loss {
+				reg-map = <0xA6 (1 << 7)>; /* reg mask */
+				gpio-num = <21>;
+			};
+			sfp07_gpio02_sfp_pres {
+				reg-map = <0xC0 (1 << 7)>; /* reg mask */
+				gpio-num = <22>;
+			};
+			sfp07_gpio02_tx_dis {
+				reg-map = <0xC6 (1 << 7)>; /* reg mask */
+				gpio-num = <23>;
+			};
+
+			/* sfp8 */
+			sfp08_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 0)>; /* reg mask */
+				gpio-num = <24>;
+			};
+			sfp08_gpio01_pres {
+				reg-map = <0xC1 (1 << 0)>; /* reg mask */
+				gpio-num = <25>;
+			};
+			sfp08_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 0)>; /* reg mask */
+				gpio-num = <26>;
+			};
+
+			/* sfp9 */
+			sfp09_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 1)>; /* reg mask */
+				gpio-num = <27>;
+			};
+			sfp09_gpio01_pres {
+				reg-map = <0xC1 (1 << 1)>; /* reg mask */
+				gpio-num = <28>;
+			};
+			sfp09_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 1)>; /* reg mask */
+				gpio-num = <29>;
+			};
+
+			/* sfp10 */
+			sfp10_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 2)>; /* reg mask */
+				gpio-num = <30>;
+			};
+			sfp10_gpio01_pres {
+				reg-map = <0xC1 (1 << 2)>; /* reg mask */
+				gpio-num = <31>;
+			};
+			sfp10_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 2)>; /* reg mask */
+				gpio-num = <32>;
+			};
+
+			/* sfp11 */
+			sfp11_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 3)>; /* reg mask */
+				gpio-num = <33>;
+			};
+			sfp11_gpio01_pres {
+				reg-map = <0xC1 (1 << 3)>; /* reg mask */
+				gpio-num = <34>;
+			};
+			sfp11_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 3)>; /* reg mask */
+				gpio-num = <35>;
+			};
+
+			/* sfp12 */
+			sfp12_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 4)>; /* reg mask */
+				gpio-num = <36>;
+			};
+			sfp12_gpio01_pres {
+				reg-map = <0xC1 (1 << 4)>; /* reg mask */
+				gpio-num = <37>;
+			};
+			sfp12_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 4)>; /* reg mask */
+				gpio-num = <38>;
+			};
+
+			/* sfp13 */
+			sfp13_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 5)>; /* reg mask */
+				gpio-num = <39>;
+			};
+			sfp13_gpio01_pres {
+				reg-map = <0xC1 (1 << 5)>; /* reg mask */
+				gpio-num = <40>;
+			};
+			sfp13_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 5)>; /* reg mask */
+				gpio-num = <41>;
+			};
+
+			/* sfp14 */
+			sfp14_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 6)>; /* reg mask */
+				gpio-num = <42>;
+			};
+			sfp14_gpio01_pres {
+				reg-map = <0xC1 (1 << 6)>; /* reg mask */
+				gpio-num = <43>;
+			};
+			sfp14_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 6)>; /* reg mask */
+				gpio-num = <44>;
+			};
+
+			/* sfp15 */
+			sfp15_gpio00_sfp_loss {
+				reg-map = <0xA7 (1 << 7)>; /* reg mask */
+				gpio-num = <45>;
+			};
+			sfp15_gpio01_pres {
+				reg-map = <0xC1 (1 << 7)>; /* reg mask */
+				gpio-num = <46>;
+			};
+			sfp15_gpio02_tx_dis {
+				reg-map = <0xC7 (1 << 7)>; /* reg mask */
+				gpio-num = <47>;
+			};
+
+			/* sfp16 */
+			sfp16_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 0)>; /* reg mask */
+				gpio-num = <48>;
+			};
+			sfp16_gpio01_pres {
+				reg-map = <0xC2 (1 << 0)>; /* reg mask */
+				gpio-num = <49>;
+			};
+			sfp16_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 0)>; /* reg mask */
+				gpio-num = <50>;
+			};
+
+			/* sfp17 */
+			sfp17_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 1)>; /* reg mask */
+				gpio-num = <51>;
+			};
+			sfp17_gpio01_pres {
+				reg-map = <0xC2 (1 << 1)>; /* reg mask */
+				gpio-num = <52>;
+			};
+			sfp17_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 1)>; /* reg mask */
+				gpio-num = <53>;
+			};
+
+			/* sfp18 */
+			sfp18_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 2)>; /* reg mask */
+				gpio-num = <54>;
+			};
+			sfp18_gpio01_pres {
+				reg-map = <0xC2 (1 << 2)>; /* reg mask */
+				gpio-num = <55>;
+			};
+			sfp18_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 2)>; /* reg mask */
+				gpio-num = <56>;
+			};
+
+			/* sfp19 */
+			sfp19_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 3)>; /* reg mask */
+				gpio-num = <57>;
+			};
+			sfp19_gpio01_pres {
+				reg-map = <0xC2 (1 << 3)>; /* reg mask */
+				gpio-num = <58>;
+			};
+			sfp19_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 3)>; /* reg mask */
+				gpio-num = <59>;
+			};
+
+			/* sfp20 */
+			sfp20_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 4)>; /* reg mask */
+				gpio-num = <60>;
+			};
+			sfp20_gpio01_pres {
+				reg-map = <0xC2 (1 << 4)>; /* reg mask */
+				gpio-num = <61>;
+			};
+			sfp20_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 4)>; /* reg mask */
+				gpio-num = <62>;
+			};
+
+			/* sfp21 */
+			sfp21_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 5)>; /* reg mask */
+				gpio-num = <63>;
+			};
+			sfp21_gpio01_pres {
+				reg-map = <0xC2 (1 << 5)>; /* reg mask */
+				gpio-num = <64>;
+			};
+			sfp21_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 5)>; /* reg mask */
+				gpio-num = <65>;
+			};
+
+			/* sfp22 */
+			sfp22_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 6)>; /* reg mask */
+				gpio-num = <66>;
+			};
+			sfp22_gpio01_pres {
+				reg-map = <0xC2 (1 << 6)>; /* reg mask */
+				gpio-num = <67>;
+			};
+			sfp22_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 6)>; /* reg mask */
+				gpio-num = <68>;
+			};
+
+			/* sfp23 */
+			sfp23_gpio00_sfp_loss {
+				reg-map = <0xA8 (1 << 7)>; /* reg mask */
+				gpio-num = <69>;
+			};
+			sfp23_gpio01_pres {
+				reg-map = <0xC2 (1 << 7)>; /* reg mask */
+				gpio-num = <70>;
+			};
+			sfp23_gpio02_tx_dis {
+				reg-map = <0xC8 (1 << 7)>; /* reg mask */
+				gpio-num = <71>;
+			};
+
+			/* sfp24 */
+			sfp24_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 0)>; /* reg mask */
+				gpio-num = <72>;
+			};
+			sfp24_gpio01_pres {
+				reg-map = <0xC3 (1 << 0)>; /* reg mask */
+				gpio-num = <73>;
+			};
+			sfp24_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 0)>; /* reg mask */
+				gpio-num = <74>;
+			};
+
+			/* sfp25 */
+			sfp25_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 1)>; /* reg mask */
+				gpio-num = <75>;
+			};
+			sfp25_gpio01_pres {
+				reg-map = <0xC3 (1 << 1)>; /* reg mask */
+				gpio-num = <76>;
+			};
+			sfp25_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 1)>; /* reg mask */
+				gpio-num = <77>;
+			};
+
+			/* sfp26 */
+			sfp26_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 2)>; /* reg mask */
+				gpio-num = <78>;
+			};
+			sfp26_gpio01_pres {
+				reg-map = <0xC3 (1 << 2)>; /* reg mask */
+				gpio-num = <79>;
+			};
+			sfp26_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 2)>; /* reg mask */
+				gpio-num = <80>;
+			};
+
+			/* sfp27 */
+			sfp27_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 3)>; /* reg mask */
+				gpio-num = <81>;
+			};
+			sfp27_gpio01_pres {
+				reg-map = <0xC3 (1 << 3)>; /* reg mask */
+				gpio-num = <82>;
+			};
+			sfp27_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 3)>; /* reg mask */
+				gpio-num = <83>;
+			};
+
+			/* sfp28 */
+			sfp28_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 4)>; /* reg mask */
+				gpio-num = <84>;
+			};
+			sfp28_gpio01_pres {
+				reg-map = <0xC3 (1 << 4)>; /* reg mask */
+				gpio-num = <85>;
+			};
+			sfp28_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 4)>; /* reg mask */
+				gpio-num = <86>;
+			};
+
+			/* sfp29 */
+			sfp29_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 5)>; /* reg mask */
+				gpio-num = <87>;
+			};
+			sfp29_gpio01_pres {
+				reg-map = <0xC3 (1 << 5)>; /* reg mask */
+				gpio-num = <88>;
+			};
+			sfp29_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 5)>; /* reg mask */
+				gpio-num = <89>;
+			};
+
+			/* sfp30 */
+			sfp30_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 6)>; /* reg mask */
+				gpio-num = <90>;
+			};
+			sfp30_gpio01_pres {
+				reg-map = <0xC3 (1 << 6)>; /* reg mask */
+				gpio-num = <91>;
+			};
+			sfp30_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 6)>; /* reg mask */
+				gpio-num = <92>;
+			};
+
+			/* sfp31 */
+			sfp31_gpio00_sfp_loss {
+				reg-map = <0xA9 (1 << 7)>; /* reg mask */
+				gpio-num = <93>;
+			};
+			sfp31_gpio01_pres {
+				reg-map = <0xC3 (1 << 7)>; /* reg mask */
+				gpio-num = <94>;
+			};
+			sfp31_gpio02_tx_dis {
+				reg-map = <0xC9 (1 << 7)>; /* reg mask */
+				gpio-num = <95>;
+			};
+
+			/* sfp32 */
+			sfp32_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 0)>; /* reg mask */
+				gpio-num = <96>;
+			};
+			sfp32_gpio01_pres {
+				reg-map = <0xC4 (1 << 0)>; /* reg mask */
+				gpio-num = <97>;
+			};
+			sfp32_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 0)>; /* reg mask */
+				gpio-num = <98>;
+			};
+
+			/* sfp33 */
+			sfp33_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 1)>; /* reg mask */
+				gpio-num = <99>;
+			};
+			sfp33_gpio01_pres {
+				reg-map = <0xC4 (1 << 1)>; /* reg mask */
+				gpio-num = <100>;
+			};
+			sfp33_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 1)>; /* reg mask */
+				gpio-num = <101>;
+			};
+
+			/* sfp34 */
+			sfp34_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 2)>; /* reg mask */
+				gpio-num = <102>;
+			};
+			sfp34_gpio01_pres {
+				reg-map = <0xC4 (1 << 2)>; /* reg mask */
+				gpio-num = <103>;
+			};
+			sfp34_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 2)>; /* reg mask */
+				gpio-num = <104>;
+			};
+
+			/* sfp35 */
+			sfp35_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 3)>; /* reg mask */
+				gpio-num = <105>;
+			};
+			sfp35_gpio01_pres {
+				reg-map = <0xC4 (1 << 3)>; /* reg mask */
+				gpio-num = <106>;
+			};
+			sfp35_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 3)>; /* reg mask */
+				gpio-num = <107>;
+			};
+
+			/* sfp36 */
+			sfp36_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 4)>; /* reg mask */
+				gpio-num = <108>;
+			};
+			sfp36_gpio01_pres {
+				reg-map = <0xC4 (1 << 4)>; /* reg mask */
+				gpio-num = <109>;
+			};
+			sfp36_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 4)>; /* reg mask */
+				gpio-num = <110>;
+			};
+
+			/* sfp37 */
+			sfp37_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 5)>; /* reg mask */
+				gpio-num = <111>;
+			};
+			sfp37_gpio01_pres {
+				reg-map = <0xC4 (1 << 5)>; /* reg mask */
+				gpio-num = <112>;
+			};
+			sfp37_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 5)>; /* reg mask */
+				gpio-num = <113>;
+			};
+
+			/* sfp38 */
+			sfp38_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 6)>; /* reg mask */
+				gpio-num = <114>;
+			};
+			sfp38_gpio01_pres {
+				reg-map = <0xC4 (1 << 6)>; /* reg mask */
+				gpio-num = <115>;
+			};
+			sfp38_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 6)>; /* reg mask */
+				gpio-num = <116>;
+			};
+
+			/* sfp39 */
+			sfp39_gpio00_sfp_loss {
+				reg-map = <0xAA (1 << 7)>; /* reg mask */
+				gpio-num = <117>;
+			};
+			sfp39_gpio01_pres {
+				reg-map = <0xC4 (1 << 7)>; /* reg mask */
+				gpio-num = <118>;
+			};
+			sfp39_gpio02_tx_dis {
+				reg-map = <0xCA (1 << 7)>; /* reg mask */
+				gpio-num = <119>;
+			};
+
+			/* sfp40 */
+			sfp40_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 0)>; /* reg mask */
+				gpio-num = <120>;
+			};
+			sfp40_gpio01_pres {
+				reg-map = <0xC5 (1 << 0)>; /* reg mask */
+				gpio-num = <121>;
+			};
+			sfp40_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 0)>; /* reg mask */
+				gpio-num = <122>;
+			};
+
+			/* sfp41 */
+			sfp41_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 1)>; /* reg mask */
+				gpio-num = <123>;
+			};
+			sfp41_gpio01_pres {
+				reg-map = <0xC5 (1 << 1)>; /* reg mask */
+				gpio-num = <124>;
+			};
+			sfp41_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 1)>; /* reg mask */
+				gpio-num = <125>;
+			};
+
+			/* sfp42 */
+			sfp42_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 2)>; /* reg mask */
+				gpio-num = <126>;
+			};
+			sfp42_gpio01_pres {
+				reg-map = <0xC5 (1 << 2)>; /* reg mask */
+				gpio-num = <127>;
+			};
+			sfp42_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 2)>; /* reg mask */
+				gpio-num = <128>;
+			};
+
+			/* sfp43 */
+			sfp43_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 3)>; /* reg mask */
+				gpio-num = <129>;
+			};
+			sfp43_gpio01_pres {
+				reg-map = <0xC5 (1 << 3)>; /* reg mask */
+				gpio-num = <130>;
+			};
+			sfp43_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 3)>; /* reg mask */
+				gpio-num = <131>;
+			};
+
+			/* sfp44 */
+			sfp44_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 4)>; /* reg mask */
+				gpio-num = <132>;
+			};
+			sfp44_gpio01_pres {
+				reg-map = <0xC5 (1 << 4)>; /* reg mask */
+				gpio-num = <133>;
+			};
+			sfp44_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 4)>; /* reg mask */
+				gpio-num = <134>;
+			};
+
+			/* sfp45 */
+			sfp45_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 5)>; /* reg mask */
+				gpio-num = <135>;
+			};
+			sfp45_gpio01_pres {
+				reg-map = <0xC5 (1 << 5)>; /* reg mask */
+				gpio-num = <136>;
+			};
+			sfp45_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 5)>; /* reg mask */
+				gpio-num = <137>;
+			};
+
+			/* sfp46 */
+			sfp46_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 6)>; /* reg mask */
+				gpio-num = <138>;
+			};
+			sfp46_gpio01_pres {
+				reg-map = <0xC5 (1 << 6)>; /* reg mask */
+				gpio-num = <139>;
+			};
+			sfp46_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 6)>; /* reg mask */
+				gpio-num = <140>;
+			};
+
+			/* sfp47 */
+			sfp47_gpio00_sfp_loss {
+				reg-map = <0xAB (1 << 7)>; /* reg mask */
+				gpio-num = <141>;
+			};
+			sfp47_gpio01_pres {
+				reg-map = <0xC5 (1 << 7)>; /* reg mask */
+				gpio-num = <142>;
+			};
+			sfp47_gpio02_tx_dis {
+				reg-map = <0xCB (1 << 7)>; /* reg mask */
+				gpio-num = <143>;
+			};
+		};
+	};
+};
+
+&ap_pinctrl {
+	ap_spi0_pins: spi-pins-0 {
+		marvell,pins = "mpp0", "mpp1", "mpp2", "mpp3";
+		marvell,function = "spi0";
+	};
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_i2c1 {
+    status = "okay";
+    clock-frequency = <100000>;
+
+	i2cmux@71 {
+		compatible = "nxp,pca9548";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp0: i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp1: i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp2: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp3: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp4: i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp5: i2c@5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp6: i2c@6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp7: i2c@7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+
+	i2cmux@72 {
+		compatible = "nxp,pca9548";
+		reg = <0x72>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp8: i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp9: i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp10: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp11: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp12: i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp13: i2c@5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp14: i2c@6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp15: i2c@7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+
+	i2cmux@73 {
+		compatible = "nxp,pca9548";
+		reg = <0x73>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp16: i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp17: i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp18: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp19: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp20: i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp21: i2c@5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp22: i2c@6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp23: i2c@7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+
+	i2cmux@74 {
+		compatible = "nxp,pca9548";
+		reg = <0x74>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp24: i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp25: i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp26: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp27: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp28: i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp29: i2c@5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp30: i2c@6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp31: i2c@7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+
+	i2cmux@75 {
+		compatible = "nxp,pca9548";
+		reg = <0x75>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp32: i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp33: i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp34: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp35: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp36: i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp37: i2c@5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp38: i2c@6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp39: i2c@7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+
+	i2cmux@76 {
+		compatible = "nxp,pca9548";
+		reg = <0x76>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp40: i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp41: i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp42: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp43: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp44: i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp45: i2c@5 {
+			reg = <5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp46: i2c@6 {
+			reg = <6>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp47: i2c@7 {
+			reg = <7>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+	pinctrl-0 = <&ap_spi0_pins>;
+	pinctrl-names = "default";
+
+
+    spidev: tpm@0 {
+       compatible = "tcg,tpm_tis-spi";
+       reg =<0>;
+       #address-cells = <1>;
+       #size-cells = <1>;
+       spi-max-frequency = <500000>;
+    };
+
+};
+
+&uart0 {
+	status = "okay";
+	pinctrl-0 = <&uart0_pins>;
+	pinctrl-names = "default";
+};
+
+
+&cp0_pinctrl {
+        cp0_ge_mdio_pins: ge-mdio-pins {
+                marvell,pins = "mpp0", "mpp1";
+                marvell,function = "ge";
+        };
+        cp0_sata0_prsnt_pin: sata0-prsnt-pins {
+                marvell,pins = "mpp29";
+                marvell,function = "sata0";
+        };
+        cp0_i2c0_pins: i2c0-pins {
+                marvell,pins = "mpp37", "mpp38";
+                marvell,function = "i2c0";
+        };
+        cp0_i2c1_pins: i2c1-pins {
+                marvell,pins = "mpp35", "mpp36";
+                marvell,function = "i2c1";
+        };
+        cp0_spi0_pins: spi0-pins {
+                marvell,pins = "mpp56", "mpp57", "mpp58", "mpp59";
+                marvell,function = "spi0";
+        };
+};
+
+&cp0_pcie0 {
+	ranges = <0x81000000 0x0 0xfb000000 0x0 0xfb000000 0x0 0xf0000
+		0x82000000 0x0 0xf6000000 0x0 0xf6000000 0x0 0x2000000
+		0x82000000 0x0 0xf9000000 0x0 0xf9000000 0x0 0x100000>;
+	phys = <&cp0_comphy0 0>;
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+
+	expander0: pca9555@21 {
+		compatible = "nxp,pca9555";
+		pinctrl-names = "default";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x21>;
+	};
+
+	eeprom_at24: at24@56 {
+		compatible = "atmel,24c64";
+		reg = <0x56>;
+	};
+
+	eeprom0: eeprom@50 {
+		compatible = "atmel,24c64";
+		reg = <0x50>;
+		pagesize = <0x20>;
+	};
+
+	eeprom1: eeprom@57 {
+		compatible = "atmel,24c64";
+		reg = <0x57>;
+		pagesize = <0x20>;
+	};
+};
+
+&cp0_nand_controller {
+	pinctrl-names = "default";
+	pinctrl-0 = <&nand_pins &nand_rb>;
+
+	nand@0 {
+		reg = <0>;
+		label = "main-storage";
+		nand-rb = <0>;
+		nand-ecc-mode = "hw";
+		nand-on-flash-bbt;
+		nand-ecc-strength = <8>;
+		nand-ecc-step-size = <512>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "U-Boot";
+				reg = <0 0x200000>;
+			};
+			partition@200000 {
+				label = "Linux";
+				reg = <0x200000 0xd00000>;
+			};
+			partition@1000000 {
+				label = "Filesystem";
+				reg = <0x1000000 0x3f000000>;
+			};
+		};
+	};
+};
+
+&cp0_spi0 {
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_spi0_pins>;
+	status = "okay";
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <108000000>;
+
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				reg = <0x00000000 0x00200000>;
+				label = "uboot";
+			};
+
+			partition@1 {
+				reg = <0x00200000 0x00010000>;
+				label = "uboot-env";
+				env_size = <0x00010000>;
+			};
+
+			partition@2 {
+				reg = <0x00210000 0x00df0000>;
+				label = "onie";
+			};
+		};
+	};
+};
+
+&cp0_sata0 {
+	status = "okay";
+
+	sata-port@1 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy1 0>;
+	};
+};
+
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+	status = "okay";
+};
+
+&cp0_mdio {
+	status = "okay";
+	OOB_E1512_PHY: ethernet-phy@1 {
+                reg = <0x0>;
+        };
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "disable";
+	/* Network PHY */
+	phys = <&cp0_comphy2 0>;
+	phy-mode = "2500base-x";
+	phy-speed=<2500>;
+	/* Generic PHY, providing serdes lanes */
+	fixed-link {
+	speed = <2500>;
+	full-duplex;
+	};
+};
+
+&cp0_eth1 {
+	status = "disable";
+	/* Network PHY */
+	phys = <&cp0_comphy3 1>;
+	phy-mode = "1000base-x";
+	fixed-link {
+	      speed = <1000>;
+	      full-duplex;
+	};
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy = <&OOB_E1512_PHY>;
+	phys = <&cp0_comphy5 2>;
+};
+
+&cp0_crypto {
+	status = "okay";
+};
