diff --git a/arch/arm64/boot/dts/marvell/accton-as4564-26p.dts b/arch/arm64/boot/dts/marvell/accton-as4564-26p.dts
new file mode 100644
index 000000000..9852aa3b6
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/accton-as4564-26p.dts
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include "armada-7040.dtsi"
+
+/ {
+	model = "Accton Marvell Armada 7040 board setup";
+	compatible = "marvell,armada-7040-as4564",
+		     "marvell,armada7040", "marvell,armada-ap806-quad",
+		     "marvell,armada-ap806";
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
+		i2c0 = &i2c0;
+		i2c1 = &cp0_i2c0;
+		i2c2 = &cp0_i2c1;
+	};
+
+	switch-cpu {
+		compatible = "marvell,prestera-switch-rxtx-sdma";
+		status = "okay";
+	};
+
+	onie_eeprom: onie-eeprom {
+		compatible = "onie-nvmem-cells";
+		status = "okay";
+		nvmem = <&eeprom_at24>;
+	};
+
+	prestera {
+		compatible = "marvell,prestera";
+		status = "okay";
+		base-mac-provider = <&onie_eeprom>;
+		ports {
+			port25 {
+				prestera,port-num = <25>;
+				sfp = <&sfp25>;
+			};
+			port26 {
+				prestera,port-num = <26>;
+				sfp = <&sfp26>;
+			};
+		};
+	};
+
+	sfp25: sfp-25 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp0>;
+
+		/*
+			The below three lines is intended to support the SFP cage interfaces:
+				los-gpio - RX_LOSS
+				mod-def0-gpio - PRESENT_BIT
+				tx-disable-gpio - TX_ENABLE (or DISABLE)
+			To support these three signal the driver 'gpio_i2c' (Or compatible platform driver) must be enabled and supported.
+			Please refer to the 'gpio_i2c' section on this DTS file below.
+			It is currently not required for Amazon 'ethtool -m' support but it is intended for future use.
+			Can be skipped in this stage.
+		*/
+		los-gpio = <&gpio_i2c 0 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 1 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 2 GPIO_ACTIVE_HIGH>;
+		maximum-power-milliwatt = <2000>;
+	};
+	sfp26: sfp-26 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c1_sfp1>;
+
+		los-gpio = <&gpio_i2c 3 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio_i2c 4 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio_i2c 5 GPIO_ACTIVE_HIGH>;
+		maximum-power-milliwatt = <2000>;
+	};
+
+	/*
+		This driver is intended to emulate GPIO interfaces on CPLD logic accessed via I2C interface.
+		It is required only if the SFP Cage interfaces will be required:
+				los-gpio - RX_LOSS
+				mod-def0-gpio - PRESENT_BIT
+				tx-disable-gpio - TX_ENABLE (or DISABLE)
+		This driver is added as reference code and can be replaced with another Platform Drivers that enables similar mapping.
+		It is currently not required for Amazon 'ethtool -m' support but it is intended for future use.
+		Can be skipped in this stage.
+	*/
+	gpio_i2c: gpio-i2c/* @40 */ {
+		compatible = "gpio-i2c-as4564";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		/* reg = <0x40>; */  /* CPLD/MUX I2C address */
+
+		gpio-map {
+			/* sfp0 */
+			sfp00_gpio00_loss {
+				reg-map = <0x40 (1 << 2)>; /* 0x40=register in the CPLD, (1 << 0)=Selected bit */
+				gpio-num = <0>;  /* Logical number used by: los-gpio, mod-def0-gpio and tx-disable-gpio */
+			};
+			sfp00_gpio01_pres {
+				reg-map = <0x41 (1 << 2)>; /* reg mask */
+				gpio-num = <1>;
+			};
+			sfp00_gpio02_tx_dis {
+				reg-map = <0x42 (1 << 2)>; /* reg mask */
+				gpio-num = <2>;
+			};
+
+			/* sfp1 */
+			sfp01_gpio00_sfp_loss {
+				reg-map = <0x40 (1 << 3)>; /* reg mask */
+				gpio-num = <3>;
+			};
+			sfp01_gpio01_sfp_pres {
+				reg-map = <0x41 (1 << 3)>; /* reg mask */
+				gpio-num = <4>;
+			};
+			sfp01_gpio02_tx_dis {
+				reg-map = <0x42 (1 << 3)>; /* reg mask */
+				gpio-num = <5>;
+			};
+		};
+	};
+};
+
+&ap_pinctrl {
+	ap_spi0_pins: spi-pins-0 {
+		marvell,pins = "mpp0", "mpp1", "mpp2", "mpp3";
+		marvell,function = "spi0";
+	};
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_i2c1 {
+    status = "okay";
+    clock-frequency = <100000>;
+
+	i2cmux@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-mux-idle-disconnect;
+
+		i2c1_sfp0: i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+		i2c1_sfp1: i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+	pinctrl-0 = <&ap_spi0_pins>;
+	pinctrl-names = "default";
+
+
+    spidev: tpm@0 {
+       compatible = "tcg,tpm_tis-spi";
+       reg =<0>;
+       #address-cells = <1>;
+       #size-cells = <1>;
+       spi-max-frequency = <500000>;
+    };
+
+};
+
+&uart0 {
+	status = "okay";
+	pinctrl-0 = <&uart0_pins>;
+	pinctrl-names = "default";
+};
+
+
+&cp0_pinctrl {
+        cp0_ge_mdio_pins: ge-mdio-pins {
+                marvell,pins = "mpp0", "mpp1";
+                marvell,function = "ge";
+        };
+        cp0_sata0_prsnt_pin: sata0-prsnt-pins {
+                marvell,pins = "mpp29";
+                marvell,function = "sata0";
+        };
+        cp0_i2c0_pins: i2c0-pins {
+                marvell,pins = "mpp37", "mpp38";
+                marvell,function = "i2c0";
+        };
+        cp0_i2c1_pins: i2c1-pins {
+                marvell,pins = "mpp35", "mpp36";
+                marvell,function = "i2c1";
+        };
+        cp0_spi0_pins: spi0-pins {
+                marvell,pins = "mpp56", "mpp57", "mpp58", "mpp59";
+                marvell,function = "spi0";
+        };
+};
+
+&cp0_pcie0 {
+	ranges = <0x81000000 0x0 0xfb000000 0x0 0xfb000000 0x0 0xf0000
+		0x82000000 0x0 0xf6000000 0x0 0xf6000000 0x0 0x2000000
+		0x82000000 0x0 0xf9000000 0x0 0xf9000000 0x0 0x100000>;
+	phys = <&cp0_comphy0 0>;
+	status = "okay";
+};
+
+&cp0_i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+
+	expander0: pca9555@21 {
+		compatible = "nxp,pca9555";
+		pinctrl-names = "default";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x21>;
+	};
+
+	eeprom_at24: at24@56 {
+		compatible = "atmel,24c64";
+		reg = <0x56>;
+	};
+
+	eeprom0: eeprom@50 {
+		compatible = "atmel,24c64";
+		reg = <0x50>;
+		pagesize = <0x20>;
+	};
+
+	eeprom1: eeprom@57 {
+		compatible = "atmel,24c64";
+		reg = <0x57>;
+		pagesize = <0x20>;
+	};
+};
+
+&cp0_nand_controller {
+	pinctrl-names = "default";
+	pinctrl-0 = <&nand_pins &nand_rb>;
+
+	nand@0 {
+		reg = <0>;
+		label = "main-storage";
+		nand-rb = <0>;
+		nand-ecc-mode = "hw";
+		nand-on-flash-bbt;
+		nand-ecc-strength = <8>;
+		nand-ecc-step-size = <512>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "U-Boot";
+				reg = <0 0x200000>;
+			};
+			partition@200000 {
+				label = "Linux";
+				reg = <0x200000 0xd00000>;
+			};
+			partition@1000000 {
+				label = "Filesystem";
+				reg = <0x1000000 0x3f000000>;
+			};
+		};
+	};
+};
+
+&cp0_spi0 {
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_spi0_pins>;
+	status = "okay";
+	spi-flash@0 {
+		#address-cells = <0x1>;
+		#size-cells = <0x1>;
+		compatible = "jedec,spi-nor";
+		reg = <0x0>;
+		spi-max-frequency = <108000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				reg = <0x00000000 0x00200000>;
+				label = "uboot";
+			};
+
+			partition@1 {
+				reg = <0x00200000 0x00010000>;
+				label = "uboot-env";
+				env_size = <0x00010000>;
+			};
+
+			partition@2 {
+				reg = <0x00210000 0x00df0000>;
+				label = "onie";
+			};
+		};
+	};
+};
+
+&cp0_sata0 {
+	status = "okay";
+
+	sata-port@1 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy1 0>;
+	};
+};
+
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	bus-width = <4>;
+	no-1-8-v;
+	non-removable;
+	status = "okay";
+};
+
+&cp0_mdio {
+	status = "okay";
+	OOB_E1512_PHY: ethernet-phy@1 {
+                reg = <0x0>;
+        };
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "disable";
+	/* Network PHY */
+	phys = <&cp0_comphy2 0>;
+	phy-mode = "2500base-x";
+	phy-speed=<2500>;
+	/* Generic PHY, providing serdes lanes */
+	fixed-link {
+	speed = <2500>;
+	full-duplex;
+	};
+};
+
+&cp0_eth1 {
+	status = "disable";
+	/* Network PHY */
+	phys = <&cp0_comphy3 1>;
+	phy-mode = "1000base-x";
+	fixed-link {
+	      speed = <1000>;
+	      full-duplex;
+	};
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy = <&OOB_E1512_PHY>;
+	phys = <&cp0_comphy5 2>;
+};
+
+&cp0_crypto {
+	status = "okay";
+};
