--- a/drivers/net/ethernet/marvell/prestera/Kconfig	2020-12-30 10:54:29.000000000 +0000
+++ b/drivers/net/ethernet/marvell/prestera/Kconfig	2022-01-06 12:50:11.946721380 +0000
@@ -24,3 +24,13 @@
 
 	  To compile this driver as a module, choose M here: the
 	  module will be called prestera_pci.
+
+config PRESTERA_SHM
+        tristate "Shared Memory interface driver for Marvell Prestera Switch ASICs family"
+        depends on PRESTERA
+        help
+          This is implementation of Shared Memory interface support for Marvell Prestera
+          Switch ASICs family.
+
+          To compile this driver as a module, choose M here: the
+          module will be called prestera_shm.
