diff --git a/arch/arm64/boot/dts/marvell/delta-tn4810m.dts b/arch/arm64/boot/dts/marvell/delta-tn4810m.dts
index 1deb7cd..13af884 100644
--- a/arch/arm64/boot/dts/marvell/delta-tn4810m.dts
+++ b/arch/arm64/boot/dts/marvell/delta-tn4810m.dts
@@ -65,8 +65,7 @@
 	};
 
 	switch-cpu {
-		compatible = "marvell,prestera-switch-rxtx-eth";
-		ethernet = <&cp0_eth0>;
+                compatible = "marvell,prestera-switch-rxtx-sdma";
 		status = "okay";
 	};
 };
diff --git a/arch/arm64/boot/dts/marvell/delta-tn48m.dts b/arch/arm64/boot/dts/marvell/delta-tn48m.dts
index 2da0eaa..6b4a9ea 100644
--- a/arch/arm64/boot/dts/marvell/delta-tn48m.dts
+++ b/arch/arm64/boot/dts/marvell/delta-tn48m.dts
@@ -65,8 +65,7 @@
         };
 
         switch-cpu {
-                compatible = "marvell,prestera-switch-rxtx-eth";
-                ethernet = <&cp0_eth0>;
+                compatible = "marvell,prestera-switch-rxtx-sdma";
                 status = "okay";
         };
 
