diff -Nur a/arch/arm64/boot/dts/marvell/delta-tn4810m.dts b/arch/arm64/boot/dts/marvell/delta-tn4810m.dts
--- a/arch/arm64/boot/dts/marvell/delta-tn4810m.dts	2020-06-15 16:47:13.746144900 +0800
+++ b/arch/arm64/boot/dts/marvell/delta-tn4810m.dts	2020-06-16 10:43:33.888740795 +0800
@@ -403,10 +403,15 @@
 			#address-cells = <1>;
 			#size-cells = <1>;
 
-			partition@0 {
-				label = "uboot";
-				reg = <0x0 0x003f0000>;
-			};
+                        partition@0 {
+                                label = "uboot";
+                                reg = <0x0 0x3f0000>;
+                        };
+
+                        partition@3f0000 {
+                                label = "uboot-env";
+                                reg = <0x3f0000 0x010000>;
+                        };
 
                         partition@400000 {
                                 label = "ONIE";
diff -Nur a/arch/arm64/boot/dts/marvell/delta-tn48m.dts b/arch/arm64/boot/dts/marvell/delta-tn48m.dts
--- a/arch/arm64/boot/dts/marvell/delta-tn48m.dts	2020-06-16 10:36:49.482109400 +0800
+++ b/arch/arm64/boot/dts/marvell/delta-tn48m.dts	2020-06-16 10:43:21.967612851 +0800
@@ -171,7 +171,12 @@
 
                         partition@0 {
                                 label = "uboot";
-                                reg = <0x0 0x400000>;
+                                reg = <0x0 0x3f0000>;
+                        };
+
+                        partition@3f0000 {
+                                label = "uboot-env";
+                                reg = <0x3f0000 0x010000>;
                         };
 
                         partition@400000 {
diff -Nur a/arch/arm64/boot/dts/marvell/delta-tx4810.dts b/arch/arm64/boot/dts/marvell/delta-tx4810.dts
--- a/arch/arm64/boot/dts/marvell/delta-tx4810.dts	2020-06-16 10:41:37.556141300 +0800
+++ b/arch/arm64/boot/dts/marvell/delta-tx4810.dts	2020-06-16 10:43:14.027920747 +0800
@@ -92,9 +92,15 @@
                         #size-cells = <1>;
 
                         partition@0 {
-                                label = "U-Boot";
-                                reg = <0x0 0x400000>;
+                                label = "uboot";
+                                reg = <0x0 0x3f0000>;
                         };
+
+                        partition@3f0000 {
+                                label = "uboot-env";
+                                reg = <0x3f0000 0x010000>;
+                        };
+
                         partition@400000 {
                                 label = "ONIE";
                                 reg = <0x400000 0xc00000>;
