--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/wnc-qsd61-aom-a-48.dts
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include "armada-7040.dtsi"
+
+/ {
+	model = "Marvell Armada 7040 WNC Avenger";
+	compatible = "marvell,armada7040-db-default", "marvell,armada7040-db",
+		     "marvell,armada7040", "marvell,armada-ap806-quad",
+		     "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+                serial0 = &uart0;
+                serial1 = &cp0_uart1;
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
+		i2c0 = &cp0_i2c0;
+		i2c1 = &cp0_i2c1;
+		i2c2 = &i2c0;
+	};
+
+	cp0_utmi: utmi@580000 {
+		compatible = "marvell,mvebu-utmi";
+		reg = <0x0 0x580000 0x0 0xc>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		cp0_utmi0: utmi@58000c {
+			compatible = "marvell,mvebu-utmi-2.6.0";
+			reg = <0x58000c 0x100>,
+			      <0x440420 0x4>,
+			      <0x440440 0x4>;
+                       utmi-port = <0> ;
+			status = "disabled";
+		};
+    };
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+
+&spi0 {
+	status = "okay";
+
+	slb9670: slb9670@0{
+		compatible = "infineon,slb9670";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>; /* CS=0 */
+		spi-max-frequency = <32000000>;
+		status = "okay";
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&cp0_uart1{
+	status = "okay";
+};
+
+&cp0_spi0 {
+           pinctrl-names = "default";
+           reg = <0x700600 0x50>;          /* control */
+	   status = "okay";
+	   spi-flash@0 {
+                     #address-cells = <0x1>;
+                     #size-cells = <0x1>;
+                     compatible = "jedec,spi-nor";
+                     reg = <0x0>;
+                     spi-max-frequency = <20000000>;
+
+                     partitions {
+                                compatible = "fixed-partitions";
+                                #address-cells = <1>;
+                                #size-cells = <1>;
+
+                                partition@0 {
+                                           label = "uboot";
+                                           reg = <0x0 0x200000>;
+                                };
+                                partition@200000 {
+                                           label = "uboot-env";
+                                           reg = <0x200000 0x10000>;
+                                           env_size = <0x10000>;
+                                };
+                                partition@210000 {
+                                           label = "onie";
+                                           reg = <0x210000 0x1400000>;
+                                };
+                                partition@1610000 {
+                                           label = "open";
+                                           reg = <0x1610000 0x9f0000>;
+                                };
+                     };
+		};
+};
+
+&cp0_pcie0 {
+	ranges = <0x81000000 0x0 0xf9020000 0x0 0xf9020000 0x0 0x10000
+		  0x82000000 0x0 0xc0000000 0x0 0xc0000000 0x0 0x30000000>;
+	status = "okay";
+	phys = <&cp0_comphy0 0>;
+};
+
+&cp0_pcie1 {
+	status = "disabled";
+	phys = <&cp0_comphy4 1>;
+};
+
+&cp0_pcie2 {
+	status = "disabled";
+};
+
+&cp0_i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_i2c1 {
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_sata0 {
+	status = "okay";
+
+	sata-port@0 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy1 0>;
+	};
+};
+
+&cp0_usb3_0 {
+	/delete-property/ usb-phy;
+	/delete-property/ phys;
+	/delete-property/ phy-names;
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&cp0_utmi0 {
+	status = "okay";
+};
+
+&cp0_mdio {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "okay";
+	/* Network PHY */
+	/*phy-mode = "sgmii";*/
+	phy-mode = "2500base-x";
+	phys = <&cp0_comphy2 0>;
+	/*phy-speed = <2500>;*/
+	/* Generic PHY, providing serdes lanes */
+	fixed-link {
+		speed = <2500>;
+		full-duplex;
+	};
+};
+
+&cp0_eth1 {
+	status = "okay";
+	/* Network PHY */
+	/*phy = <&phy0>;*/
+	phy-mode = "sgmii";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy3 1>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "sgmii";
+	phys = <&cp0_comphy5 2>;
+};
+
+&cp0_crypto {
+	status = "okay";
+};
