diff --git a/arch/arm64/boot/dts/marvell/accton-as4224.dts b/arch/arm64/boot/dts/marvell/accton-as4224.dts
new file mode 100644
index 000000000..b89e0f0ad
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/accton-as4224.dts
@@ -0,0 +1,246 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * Device Tree file for Marvell Armada 7040 Development board platform
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include "armada-7040.dtsi"
+
+/ {
+	model = "Accton Marvell Armada 7040 board setup";
+	compatible = "marvell,armada-7040-as4224",
+		     "marvell,armada7040", "marvell,armada-ap806-quad",
+		     "marvell,armada-ap806";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
+		i2c0 = &i2c0;
+		i2c1 = &cp0_i2c0;
+		i2c2 = &cp0_i2c1;
+	};
+
+	switch-cpu {
+		compatible = "marvell,prestera-switch-rxtx-sdma";
+		status = "okay";
+	};
+
+	onie_eeprom: onie-eeprom {
+		compatible = "onie-nvmem-cells";
+		status = "okay";
+		nvmem = <&eeprom_at24>;
+	};
+
+	prestera {
+		compatible = "marvell,prestera";
+		status = "okay";
+		base-mac-provider = <&onie_eeprom>;
+	};
+};
+
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&ap_pinctrl {
+                ap_spi0_pins: spi-pins-0 {
+                        marvell,pins = "mpp0", "mpp1", "mpp2",
+                        "mpp3";
+                        marvell,function = "spi0";
+		};
+        };
+
+
+&spi0 {
+	status = "okay";
+	pinctrl-0 = <&ap_spi0_pins>;
+	pinctrl-names = "default";
+
+	
+    spidev: tpm@0 {
+       compatible = "tcg,tpm_tis-spi";
+       reg =<0>; 
+       #address-cells = <1>;
+       #size-cells = <1>;
+       spi-max-frequency = <500000>; 				
+    };
+
+};
+
+&cp0_pinctrl {
+        cp0_ge_mdio_pins: ge-mdio-pins {
+                marvell,pins = "mpp0", "mpp1";
+                marvell,function = "ge";
+        };
+        cp0_sata0_prsnt_pin: sata0-prsnt-pins {
+                marvell,pins = "mpp29";
+                marvell,function = "sata0";
+        };
+        cp0_i2c0_pins: i2c0-pins {
+                marvell,pins = "mpp37", "mpp38";
+                marvell,function = "i2c0";
+        };
+        cp0_i2c1_pins: i2c1-pins {
+                marvell,pins = "mpp35", "mpp36";
+                marvell,function = "i2c1";
+        };
+        cp0_spi0_pins: spi0-pins {
+                marvell,pins = "mpp56", "mpp57", "mpp58", "mpp59";
+                marvell,function = "spi0";
+        };
+};
+
+
+&cp0_spi0 {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_spi0_pins>;
+
+        spi-flash@0 {
+                #address-cells = <0x1>;
+                #size-cells = <0x1>;
+                compatible = "jedec,spi-nor";
+                reg = <0x0>;
+                spi-max-frequency = <20000000>;
+
+                partitions {
+                        compatible = "fixed-partitions";
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+                        partition@0 {
+                                label = "U-Boot";
+                                reg = <0x0 0x400000>;
+                        };
+                        partition@400000 {
+                                label = "Filesystem";
+                                reg = <0x400000 0xc00000>;
+                        };
+                };
+        };
+
+};
+
+
+&cp0_i2c1 {
+        status = "okay";
+        clock-frequency = <100000>;
+};
+
+&spi0 {
+	status = "okay";
+	pinctrl-0 = <&ap_spi0_pins>;
+	pinctrl-names = "default";
+
+	
+    spidev: tpm@0 {
+       compatible = "tcg,tpm_tis-spi";
+       reg =<0>; 
+       #address-cells = <1>;
+       #size-cells = <1>;
+       spi-max-frequency = <500000>; 				
+    };
+
+};
+ 
+&uart0 {
+	status = "okay";
+};
+
+&cp0_pcie0 {
+       ranges = <0x81000000 0x0 0xfb000000 0x0 0xfb000000 0x0 0xf0000
+              0x82000000 0x0 0xf6000000 0x0 0xf6000000 0x0 0x2000000
+              0x82000000 0x0 0xf9000000 0x0 0xf9000000 0x0 0x100000>;
+       phys = <&cp0_comphy0 0>;
+       status = "okay";
+};
+
+
+&cp0_i2c0 {
+	status = "okay";
+	clock-frequency = <100000>;
+
+	eeprom_at24: at24@56 {
+		compatible = "atmel,24c64";
+		reg = <0x56>;
+	};
+};
+
+&cp0_sata0 {
+	status = "okay";
+
+	sata-port@1 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy1 0>;
+	};
+};
+
+&cp0_usb3_0 {
+	status = "okay";
+};
+
+&cp0_usb3_1 {
+	status = "okay";
+};
+
+&ap_sdhci0 {
+	status = "okay";
+};
+
+&cp0_mdio {
+	status = "okay";	
+	OOB_E1512_PHY: ethernet-phy@1 {
+                reg = <0x0>;
+        };
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+&cp0_eth0 {
+	status = "disable";
+	/* Network PHY */
+	phys = <&cp0_comphy2 0>;
+	phy-mode = "2500base-x";
+	phy-speed=<2500>;
+	/* Generic PHY, providing serdes lanes */
+	fixed-link {
+	speed = <2500>;
+	full-duplex;
+	};
+};
+
+&cp0_eth1 {
+	status = "disable";
+	/* Network PHY */
+	phys = <&cp0_comphy3 1>;
+	phy-mode = "1000base-x";
+	fixed-link {
+	      speed = <1000>;
+	      full-duplex;
+	};
+};
+
+&cp0_eth2 {
+	status = "okay";
+	phy-mode = "sgmii";
+	phy = <&OOB_E1512_PHY>;
+	phys = <&cp0_comphy5 2>;
+};
+
+&cp0_crypto {
+	status = "okay";
+};
+
