diff --git a/drivers/net/phy/phylink.c b/drivers/net/phy/phylink.c
index 6e66b8e77ec7..790291162df9 100644
--- a/drivers/net/phy/phylink.c
+++ b/drivers/net/phy/phylink.c
@@ -1791,6 +1791,8 @@ static int phylink_sfp_config(struct phylink *pl, u8 mode,
 		return ret;
 	}
 
+	pl->link_config.an_enabled = phylink_test(pl->supported, Autoneg);
+
 	phylink_dbg(pl, "requesting link mode %s/%s with support %*pb\n",
 		    phylink_an_mode_str(mode), phy_modes(config.interface),
 		    __ETHTOOL_LINK_MODE_MASK_NBITS, support);
@@ -1844,7 +1846,14 @@ static int phylink_sfp_module_insert(void *upstream,
 
 	return phylink_sfp_config(pl, MLO_AN_INBAND, support, support);
 }
+static void phylink_sfp_module_remove (void *upstream)
+{
+    struct phylink *pl = upstream;
 
+    ASSERT_RTNL();
+
+    linkmode_zero(pl->supported);
+}
 static int phylink_sfp_module_start(void *upstream)
 {
 	struct phylink *pl = upstream;
@@ -1949,6 +1958,7 @@ static const struct sfp_upstream_ops sfp_phylink_ops = {
 	.attach = phylink_sfp_attach,
 	.detach = phylink_sfp_detach,
 	.module_insert = phylink_sfp_module_insert,
+	.module_remove = phylink_sfp_module_remove,
 	.module_start = phylink_sfp_module_start,
 	.module_stop = phylink_sfp_module_stop,
 	.link_up = phylink_sfp_link_up,
